cfgen/cfgen

```diff
@@ -17,10 +17,48 @@ import sys
 from typing import (Any, Callable, Dict, Iterable, Iterator, List, NamedTuple,
                     Optional, Set, Tuple)
 import yaml
+import logging
 
 
 __version__ = '0.14'
 
+LOG = logging.getLogger('cfgen')
+
+
+def repeat_on_cmd_timeout(f, max_retries=-1):
```

Comment:
@suvratjoshi The logic of the decorator is good but I'm pretty sure that the developer will not be able to use `max_retries`.

In order to provide parameters to your decorator, the decorator will have to be applied like a function call:
```python
@my_super_decorator(param1, param2)
def some_func(foo, bar):
    ...
```
And that also means that the decorator must have one more function inside. It should return a **function that would return a decorator** rather than just returning decorator.

Please see how [`@repeat_if_fails`](https://github.com/Seagate/cortx-hare/blob/main/hax/hax/util.py#L100-L135) is defined for more details.